pkg/porter/archive_test.go

```diff
@@ -51,3 +54,45 @@ func TestArchive_Validate(t *testing.T) {
 		})
 	}
 }
+
+func TestArchive_AddImage(t *testing.T) {
+	p := NewTestPorter(t)
+	defer p.Close()
+
+	tests := []struct {
+		name          string
+		relocationMap relocation.ImageRelocationMap
+		inputImg      string
+		expectedImg   string
+		hasError      bool
+	}{
+		{"no relocation map set", nil, "image:v0.1.0", "", true},
+		{"image not found in relocation map", relocation.ImageRelocationMap{"image:v0.1.0": "image@sha256:123"}, "not-found-image:v0.2.0", "", true},
+		{"image successfully added", relocation.ImageRelocationMap{"image:v0.1.0": "image@sha256:123"}, "image:v0.1.0", "image@sha256:123", false},
+	}
+
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			baseImage := bundle.BaseImage{Image: tc.inputImg, Digest: "digest"}
+			ex := exporter{relocationMap: tc.relocationMap, imageStore: mockImageStore{t: t, expected: tc.expectedImg}}
+			err := ex.addImage(baseImage)
+			require.Equal(t, tc.hasError, err != nil)
```

Comment:
Can you switch from just checking that an error occurred to checking for a substring in the error message You can use the tests.RequireErrorContains helper. That way we can be sure that it failed for the reason we were expecting.